db/mysql/client.go

```diff
@@ -0,0 +1,131 @@
+	cfg := mysql.NewConfig()
+	cfg.Addr = fmt.Sprintf("%s:%s", viper.GetString(config.DbHost), viper.GetString(config.DbPort))
+	cfg.DBName = viper.GetString(config.DbName)
+	cfg.ParseTime = true
+	cfg.User = viper.GetString(config.DbUser)
+	cfg.Passwd = viper.GetString(config.DbPass)
+	return cfg.FormatDSN()
+}
+
+//NewMysqlClient initializes a mysql database connection.
+func NewMysqlClient(conf db.Option) (db.DataStore, error) {
+	log().Info("initializing mysql connection")
+
+	cli, err := sqlx.Connect("mysql", formatDSN())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect")
+	}
+	return &client{db: cli}, nil
+}
+
+// Registration stores new user details in user table
+func (c *client) Registration(user *models.User) error {
+	if user.ID != "" {
+		return fmt.Errorf("Invalid User ID")
+	}
+
+	user.ID = guuid.NewV4().String()
+	names := user.Names()
+	if _, err := c.db.NamedExec(fmt.Sprintf(`INSERT INTO user (%s) VALUES(%s)`, strings.Join(names, ","), strings.Join(mkPlaceHolder(names, ":", func(name, prefix string) string {
+		return prefix + name
+	}), ",")), user); err != nil {
+		return errors.Wrap(err, "failed to add user")
+	}
+
+	return nil
+}
+
+// UserLogin return password on successful call else error
+func (c *client) UserLogin(email string) (string, error) {
+	var user models.User
+	if err := c.db.Get(&user.Password, fmt.Sprintf(`SELECT password FROM user WHERE email='%s'`, email)); err != nil {
+		if sql.ErrNoRows != nil {
+			return "", errors.Wrap(err, "failed to fetch user....not found")
+		}
+
+		return "", err
+	}
+
+	return user.Password, nil
+}
+
+// ViewProfile return user details from db on successful call else error
+func (c *client) ViewProfile(email string) (*models.User, error) {
+	var user models.User
+	if err := c.db.Get(&user, fmt.Sprintf(`SELECT * FROM user WHERE email='%s'`, email)); err != nil {
+		if sql.ErrNoRows != nil {
+			return nil, errors.Wrap(err, "failed to fetch user....not found")
+		}
+		return nil, err
+	}
+
+	return &user, nil
+}
+
+// SubscribeItem add user subscribed item details on successful call else return error
+func (c *client) SubscribeItem(email string, subscription *models.Subscription) error {
+	if subscription.ID != "" {
+		return fmt.Errorf("Invalid User ID")
+	}
+	subscription.ID = guuid.NewV4().String()
+	user, _ := c.ViewProfile(email)
+	subscription.UserId = user.ID
+	names := subscription.Names()
+	if _, err := c.db.NamedExec(fmt.Sprintf(`INSERT INTO subscription (%s) VALUES(%s) `, strings.Join(names, ","), strings.Join(mkPlaceHolder(names, ":", func(name, prefix string) string {
+		return prefix + name
+	}), ",")), subscription); err != nil {
+		return errors.Wrap(err, "failed to add user")
+	}
+
+	return nil
+}
+
+// ViewSubscription return slice of given user subscriptions on successful call
+func (c *client) ViewSubscription(email string) ([]*models.Subscription, error) {
+	var subscription []*models.Subscription
+	user, _ := c.ViewProfile(email)
```

Comment:
do we need to handle error here?

Reply:
Yes, I solved in mongo... left with mysql... resolving